### ISSUE TITLE
Support for Katana GF66 11UG 

### ISSUE THREAD

**Issue by czalaviek** [BeardOverflow/msi-ec#38]
Actually msi_ec prints in dmesg:
`msi_ec: Your firmware version is not supported!`

I prepared some EC dumps and attached files:

```
# sudo od -t x1 /sys/kernel/debug/ec/ec0/io
0000000 00 80 00 00 00 00 00 00 00 00 00 00 00 00 00 00
0000020 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
0000040 00 00 00 00 00 00 00 00 0a 05 00 00 00 04 1b 0b
0000060 03 09 00 0d 01 00 50 81 2b 16 28 3c 37 02 e0 00
0000100 00 00 64 00 65 15 00 00 65 15 c5 43 9b 0b 00 00
0000120 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
0000140 00 00 00 00 00 00 00 00 32 00 37 40 49 4c 52 58
0000160 64 2b 26 2b 30 36 3c 46 55 64 08 03 03 03 03 03
0000200 35 00 37 3d 43 49 4f 54 63 2b 00 2b 30 36 3c 46
0000220 55 64 08 03 03 03 03 02 06 0f 7d 06 0a 78 45 00
0000240 31 35 38 31 45 4d 53 31 2e 31 30 37 30 36 32 38
0000260 32 30 32 32 30 39 3a 30 37 3a 30 38 00 00 00 28
0000300 00 00 01 25 00 00 00 00 00 d3 00 ba 00 00 00 00
0000320 00 00 c1 83 0d 00 05 80 00 01 00 00 00 00 00 00
0000340 e2 00 00 65 15 00 00 00 00 00 00 00 00 00 00 00
0000360 00 00 70 00 00 64 00 00 64 00 00 00 00 00 00 00
0000400
```
---
```
# isw -cp MSI_ADDRESS_DEFAULT

EC dump 
       00 01 02 03 04 05 06 07 08 09 0A 0B 0C 0D 0E 0F
000000 00 80 00 00 00 00 00 00 00 00 00 00 00 00 00 00  >................<
000010 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00  >................<
000020 00 00 00 00 00 00 00 00 0a 05 00 00 00 04 1b 0b  >................<
000030 03 09 00 0d 01 00 50 81 2b 16 28 3c 37 02 e0 00  >......P.+.(<7...<
000040 00 00 64 00 65 15 00 00 65 15 c2 43 b4 0b 00 00  >..d.e...e..C....<
000050 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00  >................<
000060 00 00 00 00 00 00 00 00 35 00 37 40 49 4c 52 58  >........5.7@ILRX<
000070 64 2b 26 2b 30 36 3c 46 55 64 08 03 03 03 03 03  >d+&+06<FUd......<
000080 38 00 37 3d 43 49 4f 54 63 2b 00 2b 30 36 3c 46  >8.7=CIOTc+.+06<F<
000090 55 64 08 03 03 03 03 02 06 0f 7d 06 0a 78 49 00  >Ud........}..xI.<
0000a0 31 35 38 31 45 4d 53 31 2e 31 30 37 30 36 32 38  >1581EMS1.1070628<
0000b0 32 30 32 32 30 39 3a 30 37 3a 30 38 00 00 00 28  >202209:07:08...(<
0000c0 00 00 01 25 00 00 00 00 00 a2 00 b5 00 00 00 00  >...%............<
0000d0 00 00 c1 83 0d 00 05 80 00 01 00 00 00 00 00 00  >................<
0000e0 e2 00 00 65 15 00 00 00 00 00 00 00 00 00 00 00  >...e............<
0000f0 00 00 70 00 00 64 00 00 64 00 00 00 00 00 00 00  >..p..d..d.......<
000100

Profile dump 
Value       set @ address        Fan mode
0x0(0)      0xf4(byte244)        Unknown

-----------CPU-----------        -----------GPU-----------
Value       set @ address        Value       set @ address
0x37(55°C)  0x6a(byte106)        0x37(55°C)  0x82(byte130)
0x40(64°C)  0x6b(byte107)        0x3d(61°C)  0x83(byte131)
0x49(73°C)  0x6c(byte108)        0x43(67°C)  0x84(byte132)
0x4c(76°C)  0x6d(byte109)        0x49(73°C)  0x85(byte133)
0x52(82°C)  0x6e(byte110)        0x4f(79°C)  0x86(byte134)
0x58(88°C)  0x6f(byte111)        0x54(84°C)  0x87(byte135)

0x26(38%)   0x72(byte114)        0x0(0%)     0x8a(byte138)
0x2b(43%)   0x73(byte115)        0x2b(43%)   0x8b(byte139)
0x30(48%)   0x74(byte116)        0x30(48%)   0x8c(byte140)
0x36(54%)   0x75(byte117)        0x36(54%)   0x8d(byte141)
0x3c(60%)   0x76(byte118)        0x3c(60%)   0x8e(byte142)
0x46(70%)   0x77(byte119)        0x46(70%)   0x8f(byte143)
0x55(85%)   0x78(byte120)        0x55(85%)   0x90(byte144)
```
[ec.txt](https://github.com/BeardOverflow/msi-ec/files/11293305/ec.txt)

`# cat /sys/kernel/debug/ec/ec0/io > ec0.bin`
[ec0.bin.txt](https://github.com/BeardOverflow/msi-ec/files/11293308/ec0.bin.txt)
`# cat /sys/kernel/debug/ec/ec0/gpe > gpe0.bin`
[gpe0.bin.txt](https://github.com/BeardOverflow/msi-ec/files/11293309/gpe0.bin.txt)

**Comment by glpnk** [comment]
Do you have Windows installed? You can use RW Everything to determine exact addresses of some settings and values.

Mapped some (similar) address to my PR #36 [(preview)](https://github.com/BeardOverflow/msi-ec/pull/36/files?short_path=47a3dc0#diff-47a3dc0bd896970c2ff3da10f6b5daa97c5b2b1e7e51e4e596ea805eba5857f6)

But some values are unknown:
+ charge_control
+ shift_mode (available shifts unknown) (default - comfort)
+ super_battery (may be unsupported)
+ fan_mode (available modes unknown) (default - auto)
+ leds (from windows)

or need research/test:
+ webcam
+ fn_win_swap
+ cooler_boost
+ cpu (default addresses, ```bs_fan_speed_address``` may be available)
+ gpu (default addresses)
+ kbd_bl (if during dump it was on max brightness)

**Comment by teackot** [comment]
We need to compare it with #39. It's also GF66, but 11UC and a different board (1581)

**Comment by teackot** [comment]
Implemented in #129.

**Comment by GokalpA** [comment]
My MSI PULSE GL66 11UGKV with RTX3070 and i7 11800H also seems to have the same EC version; however, it seems that it is not supported yet.
